lib/ansible/cli/doc.py

```diff
@@ -693,6 +742,9 @@ def run(self):
 
         super(DocCLI, self).run()
 
+        if context.CLIARGS['no_fail_on_errors'] and not context.CLIARGS['dump']:
+            raise AnsibleOptionsError("--no-fail-on-errors can only be used with --metadata-dump.")
+
```

Comment:
Assuming you mean
```diff
diff --git a/lib/ansible/cli/doc.py b/lib/ansible/cli/doc.py
index d5c57de00d..0e87699f04 100755
--- a/lib/ansible/cli/doc.py
+++ b/lib/ansible/cli/doc.py
@@ -463,7 +463,7 @@ class DocCLI(CLI, RoleMixin):
         exclusive.add_argument("--metadata-dump", action="store_true", default=False, dest='dump',
                                help='**For internal use only** Dump json metadata for all entries, ignores other options.')
 
-        self.parser.add_argument("--no-fail-on-errors", action="store_true", default=False, dest='no_fail_on_errors',
+        exclusive.add_argument("--no-fail-on-errors", action="store_true", default=False, dest='no_fail_on_errors',
                                  help='**For internal use only** Only used for --metadata-dump. '
                                       'Do not fail on errors. Report the error message in the JSON instead.')
```
, that would make it mutually exclusive with --metadata-dump. We'd need a conditional suboption for the exclusive group, but I don't know of a good way to do that with argparse. In ansible-galaxy we inspect the raw args, which is a hack, so I didn't suggest it here (but does allow showing the correct option help).

Reply:
yes, i meant to create a option group (which has options exclusive to the top option, but I get the confusion)

Reply:
I reverted the last commit that added this check. Let's try to find a good solution for this in a follow-up PR instead of block this PR on this check.

Reply:
I would work on this with some guidance from @bcoca. It seems like there are some existing TODOs for ansible-doc options anyway, and learning how to fix this would probably help me clean up ansible-galaxy options too.